version: "3.9"

services:
  item-service:
    build: ./item-service
    command: uvicorn app.main:app --reload --workers 1 --host 0.0.0.0 --port 8000
    volumes:
      - ./item-service:/usr/src/app
      - ./db-models/models:/usr/src/models
    ports:
      - 8001:8000
    environment:
      - DATABASE_URL=postgres://postgres:postgres@host.docker.internal:5433/shedimorly
      - PYTHONPATH=/usr/src
    depends_on:
      - app-db

  person-service:
    build: ./person-service
    command: uvicorn app.main:app --reload --workers 1 --host 0.0.0.0 --port 8000
    volumes:
      - ./person-service:/usr/src/app
      - ./db-models/models:/usr/src/models
    ports:
      - 8002:8000
    environment:
      - DATABASE_URL=postgres://postgres:postgres@host.docker.internal:5433/shedimorly
      - PYTHONPATH=/usr/src
    depends_on:
      - app-db

  app-db:
    image: postgres:10.4-alpine
    ports:
      - "5433:5432"
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: shedimorly
    volumes:
      - postgres-data:/data
    networks:
      - app-network

  pgadmin:
    image: dpage/pgadmin4
    depends_on:
      - app-db
    ports:
      - "8090:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: pgadmin4@pgadmin.org
      PGADMIN_DEFAULT_PASSWORD: admin
    networks:
      - app-network

  nginx-reverse-proxy:
    image: nginx:latest
    ports:
      - 8080:8080
    volumes:
      - ./nginx_config.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      - item-service
      - person-service
volumes:
  postgres-data:
networks:
  app-network:
    driver: bridge